src/context/field.ts

```diff
@@ -0,0 +1,36 @@
+import { ResponseTransformer } from '../response'
+import { jsonParse } from '../utils/internal/jsonParse'
+import { mergeRight } from '../utils/internal/mergeRight'
+import { json } from './json'
+
+const forbiddenFieldNamesList = ['', 'data', 'errors', 'exceptions'] as const
+type ForbiddenFieldNames = typeof forbiddenFieldNamesList[number]
+
+/**
+ * Set a custom field on the GraphQL mocked response.
+ * @example res(ctx.fields('customField', value))
+ * @see {@link https://mswjs.io/docs/api/context/field}
+ */
+export const field = <FieldNameType extends string, FieldValueType>(
+  fieldName: FieldNameType extends ForbiddenFieldNames ? never : FieldNameType,
+  fieldValue: FieldValueType,
+): ResponseTransformer<string> => {
+  return (res) => {
+    assertFieldNameIsValid(fieldName)
+
+    const prevBody = jsonParse(res.body) || {}
+    const nextBody = mergeRight(prevBody, { [fieldName]: fieldValue })
+
+    return json(nextBody)(res as any) as any
+  }
+}
+
+function assertFieldNameIsValid(fieldName: string) {
+  if (forbiddenFieldNamesList.includes(fieldName as ForbiddenFieldNames)) {
```

Comment:
Consider using `invariant` here:

```ts
import { devUtils } from '../utils/internals/devUtils'

invariant(
  fieldName !== '',
  devUtils.formatMessage('Failed to set a custom field on a GraphQL response: field name cannot be empty.')
)

invariant(
  fieldName !== 'data',
  devUtils.formatMessage('Failed to set a custom "%s" field on a mocked GraphQL response: forbidden field name. Did you mean to call "ctx.data()" instead?', fieldName)
)

invariant(
  fieldName !== 'errors',
  devUtils.formatMessage('Failed to set a custom "%s" field on a mocked GraphQL response: forbidden field name. Did you mean to call "ctx.errors()" instead?', fieldName)
)
```

I'd also drop the forbidden names array and tackle each scenario individually. It may seem repetitive but we gain more contextual error messages for the developer this way. 

What do you think?

Reply:
Thank you for the example, I wasn't sure how to use invariant when checking for a specific value. I see the value of more explicit error message, I will update the PR.

Reply:
I think it's really cool that, being a dev tool, MSW can be as explicit about error messages and error behaviors as it wants. Yeah, I'd generally recommend being detailed when it comes to error messages, and if we structure each individual name check separately, we can produce a separate message for each failed state.

Reply:
Ok ! So we keep it as it is ?